application/src/main/java/org/togetherjava/tjbot/commands/moderation/WhoIsCommand.java

```diff
@@ -0,0 +1,263 @@
+
+/**
+ * You ever wanted to stalk someone on Discord? Well no need to worry anymore! I got you covered,
+ * this command allows you to look up user (or member) info.
+ */
+@SuppressWarnings("ClassWithoutLogger")
+public final class WhoIsCommand extends SlashCommandAdapter {
+    private static final String USER_OPTION = "user";
+    private static final String SHOW_SERVER_SPECIFIC_INFO = "show_server_specific_info";
+
+    // Sun, December 11, 2016, 13:36:30
+    private static final DateTimeFormatter DATE_TIME_FORMAT =
+            DateTimeFormatter.ofPattern("E, MMMM d, u, HH:mm:ss");
+
+    /**
+     * Creates an instance of who is command
+     */
+    public WhoIsCommand() {
+        super("whois", "Provides info about the given user", SlashCommandVisibility.GUILD);
+
+        getData().addOption(OptionType.USER, USER_OPTION, "the user to look up", true)
+            .addOption(OptionType.BOOLEAN, SHOW_SERVER_SPECIFIC_INFO,
+                    "whenever to show info that is specific to this server, such as ", false);
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull final SlashCommandEvent event) {
+        OptionMapping userOption = Objects.requireNonNull(event.getOption(USER_OPTION),
+                "The given user option cannot be null");
+        OptionMapping showServerSpecificInfoOption = event.getOption(SHOW_SERVER_SPECIFIC_INFO);
+
+        User user = userOption.getAsUser();
+        Member member = userOption.getAsMember();
+
+
+        boolean showServerSpecificInfo = null != member && (null == showServerSpecificInfoOption
+                || showServerSpecificInfoOption.getAsBoolean());
+
+        user.retrieveProfile().flatMap(profile -> {
+            if (showServerSpecificInfo) {
+                return handleWhoIsMember(event, member, profile);
+            } else {
+                return handleWhoIsUser(event, user, profile);
+            }
+        }).queue();
+    }
+
+    @CheckReturnValue
+    private static @NotNull ReplyAction handleWhoIsUser(final @NotNull SlashCommandEvent event,
+            final @NotNull User user, final @NotNull User.Profile profile) {
+
+        StringBuilder descriptionBuilder =
+                new StringBuilder().append(handleUserIdentification(user))
+                    .append("\n**Is bot:** ")
+                    .append(user.isBot())
+                    .append(handleUserFlags(user.getFlags()))
+                    .append("\n**Registration date:** ")
+                    .append(DATE_TIME_FORMAT.format(user.getTimeCreated()));
+
+        EmbedBuilder embedBuilder =
+                generateEmbedBuilder(event, user, profile, profile.getAccentColor()).setAuthor(
+                        user.getName(), user.getEffectiveAvatarUrl(), user.getEffectiveAvatarUrl())
+                    .setDescription(descriptionBuilder);
+
+        return event.replyEmbeds(embedBuilder.build())
+            .addActionRow(Button.of(ButtonStyle.LINK, "discord://-/users/" + user.getId(),
```

Comment:
This discord-URI usage is new to me, probably also new to others. It might be worth to move that up into some more general helper class with helper methods. Some sort of `DiscordActionBuilder` or `DiscordActions` or similar maybe, with a `createShowProfileAction(id)` and stuff like that 🤷

Reply:
I'd love to add that, but preferably in another PR in the future.

There's a ton of Discord URL's (which I'd love to wrap, maybe people can do cool things with it)
But I've to check how many work, and to what extent.

Just adding that class for method feels wrong to me, I'd rather just create a PR in the future with all of the actions

Reply:
See #391